src/genie/libs/parser/linux/route.py

```diff
@@ -160,3 +153,288 @@ def cli(self, output=None):
+        # 172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1
+
+        p3 = re.compile(r'(?P<destination>[a-z0-9\.\:\/]+)'
+                         ' dev (?P<device>[a-z0-9\.\-]+)'
+                         ' proto (?P<proto>\w+)'
+                         ' scope (?P<scope>\w+)'
+                         ' src (?P<src>[a-z0-9\.\:\/]+)'
+                         )
+
+        # 172.18.0.0/16 dev br-d19b23fac393 proto kernel scope link src 172.18.0.1 linkdown 
+
+        p4 = re.compile(r'(?P<destination>[a-z0-9\.\:\/]+)'
+                         ' dev (?P<device>[a-z0-9\.\-]+)'
+                         ' proto (?P<proto>\w+)'
+                         ' scope (?P<scope>\w+)'
+                         ' src (?P<src>[a-z0-9\.\:\/]+)'
+                         ' linkdown '
+                         )       
+
+        # 192.168.1.0/24 dev enp7s0 proto kernel scope link src 192.168.1.212 metric 100
+
+        p5 = re.compile(r'(?P<destination>[a-z0-9\.\:\/]+)'
+                         ' dev (?P<device>[a-z0-9\.\-]+)'
+                         ' proto (?P<proto>\w+)'
+                         ' scope (?P<scope>\w+)'
+                         ' src (?P<src>[a-z0-9\.\:\/]+)'
+                         ' metric (?P<metric>[\d]+)'
+                         )
+
+        # broadcast 127.0.0.0 dev lo table local proto kernel scope link src 127.0.0.1 
+
+        p6 = re.compile(r'broadcast (?P<destination>[a-z0-9\.\:\/]+)'
+                         ' dev (?P<device>[a-z0-9\.\-]+)'
+                         ' table (?P<table>\w+)'
+                         ' proto (?P<proto>\w+)'
+                         ' scope (?P<scope>\w+)'
+                         ' src (?P<src>[a-z0-9\.\:\/]+)'
+                         )
+                         
+        
+        # local 10.233.44.70 dev kube-ipvs0 table local proto kernel scope host src 10.233.44.70
+
+        p7 = re.compile(r'local (?P<destination>[a-z0-9\.\:\/]+)'
+                         ' dev (?P<device>[a-z0-9\.\-]+)'
+                         ' table (?P<table>\w+)'
+                         ' proto (?P<proto>\w+)'
+                         ' scope (?P<scope>\w+)'
+                         ' src (?P<src>[a-z0-9\.\:\/]+)'
+                         )
+
+
+
+        # Initializes the Python dictionary variable
+        parsed_dict = {}
+
+
+        # Defines the "for" loop, to pattern match each line of output
+
+        for line in out.splitlines():
+            line = line.strip()
+
+            # default via 192.168.1.1 dev enp7s0 proto dhcp metric 100
+            m = p1.match(line)
+            if m:
+                if 'routes' not in parsed_dict:
+                    parsed_dict.setdefault('routes', {})
+
+                group = m.groupdict()
+                gateway = group['gateway']
+                interface = group['device']
+                metric = int(group['metric'])
+
+                
+                if gateway:
+                    parsed_dict['routes'] = { '0.0.0.0': {
```

Comment:
I think using **setdefault** to assign the dictionary values will be better like in line 299